cmd/snap/cmd_run.go

```diff
@@ -112,6 +112,16 @@ and environment.
 		}, nil)
 }
 
+// isStopping returns true if the system is shutting down.
+func isStopping() (bool, error) {
+	cmd := exec.Command("systemctl", "is-system-running")
+	stdout, err := cmd.Output()
```

Comment:
I think this needs to be more elaborate. Specifically, the manpage says the return status is > 0 for anyting else than `running`. Then, some states are still functional, eg. `degraded` but return a non-0 exit code.

```
if err == nil {
    // running
    return false, nil
}
if eErr, ok := err.(*ExitError); ok {
    // TODO: is stdout really captured in this case?
    if stdout != nil && string(stdout) == "stopping\n" {
        return true, nil
    }
    if len(eErr.stderr) == 0 {
        // systemctl(1) states that output is written to stdout, so nothing on stderr
        // means it's likely not an error
        return false, nil
    }
}
return false, err
```

Reply:
I read the man page but missed that os/exec.Cmd behaves this way. I'll adjust this.

Reply:
Done